File: fastapi-app/src/service/user_service.py
import os
from pydantic import BaseModel
import aiohttp


class User(BaseModel):
    id: str
    name: str
    created_at: int
    email: str
    password: str


async def loginUser(name: str, email: str, password: str) -> User:
    async with aiohttp.ClientSession() as session:
        async with session.post(
            f'{os.getenv("API_URL","")}/api/v1/user/login',
            data={"name": name, "email": email, "password": password},
        ) as response:
            return await response.json()

File: express-app/src/services/userService.ts
import axios from "axios";
import User from "../interfaces/User";
import Logger from "../utils/logger";

export const getUser = async (name: string, email: string, password: string): Promise<User> => {
  Logger.info("Get User");
  const { data } = await axios.post(`${process.env.API_URL}/api/v1/user/login`, {
    name,
    email,
    password,
  });
  return data;
};
